Arm/Cosim.lean

```diff
@@ -206,30 +210,52 @@ def regStates_match (input o1 o2 : regState) : IO Bool := do
      pure false
 
 /-- Run one random test for the instruction `inst`. -/
-def one_test (inst : BitVec 32) : IO Bool := do
+def one_test (inst : BitVec 32) (guid : Nat) : IO Bool := do
   let input      ← input_regState inst
-  let machine    ← arm_cosim_test input
+  let machine    ← arm_cosim_test input guid
   let machine_st := machine_to_regState inst machine
   let model      := run 1 (regState_to_armState input)
   let model_st := model_to_regState inst model
   regStates_match input machine_st model_st
 
+/--
+Make a task for running a single test.
+Use a `guid` to create unique files to ensure that tests do not trample on each other.
+Return `some t` if a test can be produced, and `none` if not.
+- Uses `IO` to try produce a random test that shall be run in a task.
+- Returns `.none` if there is the instruction does not exist on the given architecture.
+- Returns `.some task` upon succeessful creation of the task to randomly test one instruction instance.
+-/
+def mk_one_test_task (verbose : Bool) (logPrefix : String)
+    (fn : IO (Option (BitVec 32))) (guid : Nat) :
+    IO (Option (Task (Except IO.Error Bool))) := do
+  let maybe_inst ← fn
```

Comment:
This seems to create a separate `Task` per test cases. I vaguely remember a discussion that Lean's `Task` can be a bit heavy, and in particular, it's best to not create more `Task`s than CPU cores.

I'm not sure how accurate this memory is w.r.t. the current situation, this might've been quite some time ago.

Reply:
This is cleaner to implement, and in the glorious future when `Task`s will work as intended (green threads), this should perform even better than it currently does!